consensus/src/common.rs

```diff
@@ -131,14 +133,21 @@ impl Chain<ProtocolState> for ProtocolStateChain {
         self.0.len()
     }
 
-    fn last_vrf_hash(&self) -> Option<Fp> {
-        static HASHER_POOL: OnceCell<SpinLockObjectPool<PoseidonHasherKimchi<VrfOutputTruncated>>> =
-            OnceCell::new();
-        let pool =
-            HASHER_POOL.get_or_init(|| SpinLockObjectPool::new(|| create_kimchi(()), |_| ()));
-        let mut hasher = pool.pull();
-        self.top()
-            .map(|s| hasher.hash(&s.body.consensus_state.last_vrf_output))
+    fn last_vrf_hash(&self) -> Result<String, ConsensusError> {
```

Comment:
I don't think we should ever be using the `String` type for hashes. We have loads of different more informative hash types to use

Reply:
Just following the spec here: https://github.com/MinaProtocol/mina/blob/compatible/docs/specs/consensus/README.md#516-hashlastvrf